### PR TITLE
Reformat get products route

### PR DESCRIPTION
**Pull request by wenhaogoh**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,544 @@
+
```
Comment (ivantjh) [inline docs/index.md:1]: Can version control the postman collection? Makes it easier for other people to edit and generate the docs.

Comment (wenhaogoh) [reply docs/index.md:1]: Done

File: src/controllers/productController.ts [hunk 2]
```diff
@@ -56,7 +42,18 @@ async function retrieveProduct(req: Request, res: Response, next: NextFunction)
...
+          association: Product.associations.Stall,
+          include: [
+            {
+              association: Stall.associations.HawkerCentre,
+            },
+          ],
+        },
+      ],
+    });
     req.body = products;
```
Comment (ivantjh) [inline src/controllers/productController.ts]: Can just end the request by doing`res.status(200).json(products);`, no need this line and the next line.

File: src/controllers/productController.ts [hunk 3]
```diff
@@ -99,7 +96,7 @@ async function destroyProduct(req: Request, res: Response, next: NextFunction) {
   }
 }
 
-export const indexProductFuncs = [indexProduct, getMultipleIdFormatting];
+export const indexProductFuncs = [indexProduct, showProduct];
```
Comment (ivantjh) [inline src/controllers/productController.ts]: Lets not mix the individual operation calls before it gets too confusing. `showProduct` should only be present in `showProductFuncs`. With the above modification, can just do `export const indexProductFuncs = [indexProduct];`

File: src/controllers/productController.ts [hunk 4]
```diff
@@ -99,7 +96,7 @@ async function destroyProduct(req: Request, res: Response, next: NextFunction) {
...
 export const showProductFuncs = [retrieveProduct, getIdFormatting, showProduct];
 export const createProductFuncs = [postIdFormatting, createProduct];
```
Comment (ivantjh) [inline src/controllers/productController.ts]: Could probably just remove the `getIdFormatting` and `postIdFormatting` for consistency in the API responses. Can first comment out their respective functions above too.

File: src/models/Product.ts [hunk 5]
```diff
@@ -33,7 +33,6 @@ class Product
   public price!: number | null;
   public image!: string | null;
   public stallId!: number;
-
```
Comment (ivantjh) [inline src/models/Product.ts]: I left a line in the other models to differentiate between `readonly` and the normal variables.

**Review by ivantjh — APPROVED** [review]
LGTM!

**Comment by ivantjh** [comment]
Should merge yours first before @zatkiller's PR so he can document the changes to the APIs

File: src/controllers/productController.ts (hunk 2 above)
Comment (wenhaogoh) [reply src/controllers/productController.ts]: Removed

File: src/controllers/productController.ts (hunk 3 above)
Comment (wenhaogoh) [reply src/controllers/productController.ts]: Done

File: src/models/Product.ts (hunk 5 above)
Comment (wenhaogoh) [reply src/models/Product.ts]: Added

File: src/controllers/productController.ts (hunk 4 above)
Comment (wenhaogoh) [reply src/controllers/productController.ts]: Removed

**Pull request merged**